src/go/rpk/pkg/net/interfaces.go

```diff
@@ -73,3 +74,14 @@ func GetFreePortPool(n int) ([]uint, error) {
 	}
 	return ports, nil
 }
+
+func CheckIfPortIsFree(port uint) error {
+	conn, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
+	if err != nil {
+		return err
+	}
+	if conn != nil {
+		defer conn.Close()
+	}
```

Comment:
If there is no error you might as well just `defer conn.Close()` without the nil check

Reply:
Right, the function does not return a pointer. My bad as my IDE still doesn't show the the source code for packages for the repo, I have been skipping that part. :/ Will probably need to fix it soon